### ISSUE TITLE
Export SVG shadow for rectangles

### ISSUE THREAD

**Issue by Maddoc42** [paperjs/paper.js#1003]
Is there a way to generate shadows for a `Path.Rectangle` and have them exported via SVG?

The `Path.shadow*` values work great when working on a canvas, but the shadow is not  exported when running `Project.exportSVG()`. In addition `Path` does not seem to support any blurring of the fill color.

The only super hackish way I see right now, is to create a pseudo-shadow via multiple gradients ...

Thanks for the great library btw!

**Comment by iconexperience** [comment]
Exporting drop shadows to SVG is currently not supported for the simple reason that there is not equivalent drop shadow property in SVG.

Anyway, it can be done through filters, which is not really trivial, but absolutely doable. Here is how a filter would look like in "pseudo-SVG":

```
<filter>
  <feColorMatrix />  // set the color to shadow color
  <feGaussianBlur /> // create a blur with the shadow's blur extent
  <feOffset /> // offset the shadow using the shadow offset
  <feMerge in2="SourceGraphic" /> // merge the original path with result of the above filter steps
</filter>
```

This filter has to be set to each path that has a drop shadow.

**Comment by iconexperience** [comment]
I just noticed that the `feMerge` filter primitive can probably be replaced by a `feComposite`, which would make the filter structure a bit more concise. If you are interested in digging deeper into filters, please take a look at  https://www.w3.org/TR/SVG/filters.html

**Comment by lehni** [comment]
Yeah I wasn't very keen on that rather excessive structure. It's a shame it's not supported through the CSS property yet, similar to how blend modes now work. The whole SVG / CSS / HTML mix is a pretty messy affair, unfortunately.

**Comment by iconexperience** [comment]
I have not looks too deeply into the SVG 2.0 spec that should be finished by the end of this year. I heared that they will improve svg/css integration substantially, but of course if the spec is finished by the end of this year, how long will it take until all browsers will support it properly?

**Comment by lehni** [comment]
This example here is slightly different to yours: http://codepen.io/thejamespower/pen/myZMpj/

**Comment by iconexperience** [comment]
Yes, but mine is better :)

Here is why: With my filter the shadow color is set through `feColorMatrix`, while in the example above the shadow is always black (using `SourceAlpha` as input). So in the example, the `shadowColor` style property is completely ignored.

**Comment by iconexperience** [comment]
Here is a modified pen that uses my proposed implementation. The shadow color is r=0.1 g=0.3 b=0.2 a=0.8
http://codepen.io/anon/pen/NNdpEe

**Comment by lehni** [comment]
cool, thanks! we'd need a way to merge shadow definitions, using the various values that define the shadow as a hash. it'll add quite a bit of code, probably.

**Comment by lehni** [comment]
And importing those darn things again gets even more complicated.

**Comment by iconexperience** [comment]
Yes, that is next to impossible.

**Comment by Maddoc42** [comment]
Wow, thanks for the pointer to the SVG filters, greatly appreciated :) that pretty much solves the problem!

**Comment by lehni** [comment]
Let's keep it open until we decide wether to export shadows in the library or not.

**Comment by dasheck0** [comment]
@lehni So have you decided yet? :)

**Comment by pete-jones** [comment]
This would be a really useful feature to be able to export an svg with a shadow. Any further progress?
